lib/profiler.rb

```diff
@@ -0,0 +1,23 @@
+# adapted from argo
+# takes a block of code, starts the profiler, runs the code, then
+# stops the profiler and returns the results of the profiling.
+# example usage:
+#  profiler = Profiler.new
+#  profiler.prof { MoabToCatalog.seed_catalog(storage_dir) }
+#  profiler.print_results_flat(out_file_id)
+class Profiler
+
+  attr_accessor :results
+
+  def prof
+    RubyProf.start
+    yield
+    @results = RubyProf.stop
+  end
+
+  def print_results_flat(out_file_id)
+    File.open "log/#{out_file_id}-flat.txt", 'w' do |file|
```

Comment:
what about a timestamp in the filename, just so profiling runs don't step on each other?  could also append to the file, but i think i'd have a slight pref for timestamp, because then you wouldn't have one indefinitely growing file.